model_compression_toolkit/core/keras/hessian/weights_trace_hessian_calculator_keras.py

```diff
@@ -50,5 +53,112 @@ def __init__(self,
                                                                  trace_hessian_request=trace_hessian_request,
                                                                  num_iterations_for_approximation=num_iterations_for_approximation)
 
-    def compute(self):
-        Logger.error(f"Hessian trace approx w.r.t weights is not supported for now")
\ No newline at end of file
+    def compute(self) -> np.ndarray:
+        """
+        Compute the Hessian-based scores for w.r.t target node's weights.
+        Currently, supported nodes are [Conv2D, Dense, Conv2DTranspose, DepthwiseConv2D].
+        The computed scores are returned in a numpy array. The shape of the result differs
+        according to the requested granularity. If for example the node is Conv2D with a kernel
+        shape of (3, 3, 3, 2) (namely, 3 input channels, 2 output channels and kernel size of 3x3)
+        and the required granularity is HessianInfoGranularity.PER_TENSOR the result shape will be (1,),
+        for HessianInfoGranularity.PER_OUTPUT_CHANNEL the shape will be (2,) and for
+        HessianInfoGranularity.PER_ELEMENT a shape of (3, 3, 3, 2).
+
+        Returns:  The computed scores as numpy ndarray.
+
+        """
+        # Check if the target node's layer type is supported
+        if self.hessian_request.target_node.layer_class not in [Conv2D, Dense, Conv2DTranspose, DepthwiseConv2D]:
+            Logger.error(
+                f"{self.hessian_request.target_node.type} is not supported for Hessian info w.r.t weights.")
+
+        # Construct the Keras float model for inference
+        model, _ = FloatKerasModelBuilder(graph=self.graph).build_model()
+        # TODO: what to do in reuse
```

Comment:
what is the problem in reuse?

Reply:
Since some nodes are the same layer (in the framework model), computing their scores is needed only for one of them (the scores for the rest of them will be the same scores as the one we computed once). For this reason, in the HessianInfoService we'll need to add this logic that if a request contains a reused node, we'll take the computation of the "master" node (in the reuse group).